test/unit/BMath.t.sol

```diff
@@ -4,564 +4,263 @@ pragma solidity 0.8.25;
+    _weightOut = bound(_weightOut, BONE, type(uint256).max);
+    _balanceOut = bound(_balanceOut, (type(uint256).max - weightOut / 2), type(uint256).max);
 
-    // Post condition
-    // it will revert (overflow)
-    //     token balance out * BONE + (token weight out/2) > uint max
-    vm.expectRevert('ERR_DIV_INTERNAL');
+    // it should revert
+    //     bo * BONE + (wo / 2) > uint256 max
+    vm.expectRevert(BNum.BNum_DivInternal.selector);
 
-    // Action
-    bMath.calcSpotPrice(_tokenBalanceIn, _tokenWeightIn, _tokenBalanceOut, _tokenWeightOut, _swapFee);
+    bMath.calcSpotPrice(balanceIn, weightIn, _balanceOut, _weightOut, swapFee);
   }
 
   function test_CalcSpotPriceWhenUsingASwapFeeOfZero() external {
-    // Precondition
+    // it should return correct value
+    //     bi/wi * wo/bo
+    //     20/1 * 2/30 = 1.333333...
     uint256 _swapFee = 0;
 
-    uint256 _tokenWeightIn = 10 * BONE;
-    uint256 _tokenWeightOut = 20 * BONE;
-    uint256 _tokenBalanceIn = 100 * BONE;
-    uint256 _tokenBalanceOut = 30 * BONE;
+    uint256 _spotPrice = bMath.calcSpotPrice(balanceIn, weightIn, balanceOut, weightOut, _swapFee);
 
-    // expected is bi/wi / bo/wo
-    uint256 _expected = bNum.bdivExposed(
-      bNum.bdivExposed(_tokenBalanceIn, _tokenWeightIn), bNum.bdivExposed(_tokenBalanceOut, _tokenWeightOut)
-    );
-
-    // Action
-    uint256 _spotPrice =
-      bMath.calcSpotPrice(_tokenBalanceIn, _tokenWeightIn, _tokenBalanceOut, _tokenWeightOut, _swapFee);
-
-    // Post condition
-    // it should return bi/wi / bo/wo
-    assertEq(_spotPrice, _expected, 'wrong spot price when fee is 0');
+    assertEq(_spotPrice, 1.333333333333333333e18);
   }
 
   function test_CalcSpotPriceWhenUsingKnownValues() external {
-    vm.skip(true); // use a set of known in/result
-    // Precondition
-    uint256 _swapFee = BONE / 2;
-
-    uint256 _tokenWeightIn = 10 * BONE;
-    uint256 _tokenWeightOut = 20 * BONE;
-    uint256 _tokenBalanceIn = 100 * BONE;
-    uint256 _tokenBalanceOut = 30 * BONE;
-
-    // expected is bi/wi / bo/wo * 1/(1 - sf)
-    uint256 _expected = bNum.bmulExposed(
-      bNum.bdivExposed(
-        bNum.bdivExposed(_tokenBalanceIn, _tokenWeightIn), bNum.bdivExposed(_tokenBalanceOut, _tokenWeightOut)
-      ),
-      bNum.bdivExposed(BONE, bNum.bsubExposed(BONE, _swapFee))
-    );
+    // it should return correct value
+    //     (bi/wi * wo/bo) * (1 / (1 - sf))
+    //     (20/1 * 2/30) * (1 / (1 - 0.1)) = 1.481481481...
 
-    // Action
-    uint256 _spotPrice =
-      bMath.calcSpotPrice(_tokenBalanceIn, _tokenWeightIn, _tokenBalanceOut, _tokenWeightOut, _swapFee);
+    uint256 _spotPrice = bMath.calcSpotPrice(balanceIn, weightIn, balanceOut, weightOut, swapFee);
 
-    // Post condition
-    // it should return correct value
-    assertEq(_spotPrice, _expected, 'wrong spot price when fee is not 0');
+    assertEq(_spotPrice, 1.481481481481481481e18);
```

Comment:
summing up what we discussed: I think using an easily extendable array of known values and results as a complement of this would be great, but would recommend keeping one testcase outside of it

- as a mitigation on the 'runner' generating false negatives
- a way to see how broken a feature is when tests fail (some edge case in the array failing vs the trivial case)

Reply:
we can think about later, i believe this abstraction is a step on the right way tho